libraries/libstratosphere/source/settings/impl/settings_key_value_store.cpp

```diff
@@ -0,0 +1,1629 @@
+                using difference_type = ptrdiff_t;
+                using pointer         = T*;
+                using const_pointer   = const T*;
+                using reference       = T&;
+                using const_reference = const T&;
+            public:
+                Allocator() { /* ... */ }
+                ~Allocator() { /* ... */ }
+
+                pointer allocate(size_type n, const_pointer hint = 0) {
+                    AMS_UNUSED(hint);
+                    return static_cast<pointer>(AllocateFromHeap(sizeof(T) * n));
+                }
+
+                void deallocate(pointer p, size_type n) {
+                    FreeToHeap(p, sizeof(T) * n);
+                }
+        };
+        
+        os::SdkMutex g_key_value_store_mutex;
+
+        void ClearKeyValueStoreMap(Map &map) {
+            /* Free all values to the heap. */
+            for (auto it = map.begin(); it != map.end(); ++it) {
+                FreeMapValueToHeap(it->second);
+            }
+
+            /* Clear the map. */
+            map.clear();
+        }
+
+        bool CompareValue(const void *lhs, size_t lhs_size, const void *rhs, size_t rhs_size) {
+            /* Check if both buffers are the same. */
+            if (lhs == rhs) {
+                return true;
+            }
+
+            /* If the value sizes don't match, return false. */
+            if (lhs_size != rhs_size) {
+                return false;
+            }
+
+            /* If the value sizes are 0, they are considered to match. */
+            if (lhs_size == 0) {
+                return true;
+            }
+            
+            /* Compare the two values if they are non-null. */
+            return lhs != nullptr && rhs != nullptr && std::memcmp(lhs, rhs, lhs_size) == 0;
+        }
+
+        void FreeMapValueToHeap(const MapValue &map_value) {
+            /* Free the current value. */
+            if (map_value.current_value != nullptr && map_value.current_value != map_value.default_value) {
+                FreeToHeap(map_value.current_value, map_value.current_value_size);
+            }
+
+            /* Free the default value. */
+            if (map_value.default_value != nullptr) {
+                FreeToHeap(map_value.default_value, map_value.default_value_size);
+            }
+        }
+
+        void FreeToHeap(void *block, size_t size) {
+            AMS_UNUSED(size);
+            lmem::FreeToExpHeap(*GetHeapHandle(), block);
+        }
+
+        void *AllocateFromHeap(size_t size) {
+            return lmem::AllocateFromExpHeap(*GetHeapHandle(), size);
+        }
+
+        size_t GetHeapAllocatableSize() {
+            return lmem::GetExpHeapAllocatableSize(*GetHeapHandle(), alignof(s32));
+        }
+
+        lmem::HeapHandle *GetHeapHandle() {
+            static constinit bool s_is_initialized = false;
+            static constinit lmem::HeapHandle s_heap_handle;
+            static constinit u8 s_heap_memory[32_KB];
```

Comment:
This should be 512KB.

Also, you should have constexpr size_t HeapMemorySize = 512_KB; up above in earlier defs, and use it here.